src/box/iproto.cc

```diff
@@ -3000,11 +3010,24 @@ iproto_listen(const char *uri)
 	 * implementation, we rely on the Linux kernel to distribute
 	 * incoming connections across iproto threads.
 	 */
-	if (evio_service_bind(&tx_binary, uri) != 0)
-		return -1;
-	if (iproto_send_listen_msg(&tx_binary) != 0)
-		return -1;
-	return 0;
+	int rc = 0;
+	bool binary_is_old = false;
+	if (evio_service_bind(&tmp, uri) != 0) {
+try_to_bind_old_binary:
+		if (evio_service_bind_addr(&tx_binary))
+			return -1;
+		rc = -1;
+		binary_is_old = true;
+		memcpy(&tmp, &tx_binary, sizeof(struct evio_service));
```

Comment:
Copying evio_service struct with memcpy looks dangerous IMO, just like initialization with memset.

Reply:
Move this part to lua